db/db.go

```diff
@@ -2,28 +2,37 @@ package db
 
```

Comment:
Honestly, I feel like the original code was just fine. From what I've been learning about mgo, no one has mentioned a need to protect the connection logic from concurrent access. This is because they either do it in init() or main(), both of which will only be started once on program start up. Also, if this code was written in order to pass access to the database connection, it was suggested by the original writer of mgo to use Session.Copy(), which not only passes the session around, but opens a new socket (from what it sounds like) and allows queries to be executed concurrently.